### PR TITLE
Adding autopause APIs to future

### PR DESCRIPTION
**Pull request by ridv**
* Added new PauseUpdateMonitor which monitors an update until it is an `ROLL_FORWARD_PAUSED` state.
* Added variableBatchStep command to sample client to be used for testing new VariableBatchStep api.

### CODE REVIEW

File: realis.go [hunk 1]
```diff
@@ -1059,3 +1060,50 @@ func (r *realisClient) RollbackJobUpdate(key aurora.JobUpdateKey, message string
 	}
 	return resp, nil
 }
+
+// VariableBatchStep returns the current batch the update is in during a variable batch update.
+func (r *realisClient) VariableBatchStep(key aurora.JobUpdateKey) (int, error) {
+
+	// Query Aurora for an Update that is either paused or rolling forward
+	resp, err := r.JobUpdateDetails(aurora.JobUpdateQuery{
+		UpdateStatuses: []aurora.JobUpdateStatus{
```
Comment (kkrishna) [inline realis.go]: how do we handle failure path as we are filtering on ROLLING_FORWARD and ROLL_FORWARD_PAUSED

File: realis.go [hunk 2]
```diff
@@ -1059,3 +1060,50 @@ func (r *realisClient) RollbackJobUpdate(key aurora.JobUpdateKey, message string
...
+	if len(jobUpdates) == 0 {
+		return 0, errors.New("update provided could not be found in Aurora")
+	}
+
+	strategy := jobUpdates[0].GetUpdate().GetInstructions().GetSettings().GetUpdateStrategy()
+	if !strategy.IsSetVarBatchStrategy() {
+		return 0, errors.New("update strategy used by update config is not supported by this function")
+	}
+
+	// Using a map to create a set to guard against multiple INSTANCE_UPDATING events for the same shard.
```
Comment (kkrishna) [inline realis.go]: would we have multiple INSTANCE_UPDATING for a shard?

Comment (rdelval) [reply realis.go]: Not according to my tests, but guarding against any future changes and/or unwelcome surprises.

File: realis.go (hunk 1 above)
Comment (rdelval) [reply realis.go]: We return an error if there's no updates matching the statuses asked for in L1082

**Pull request merged**